### ISSUE TITLE
Analyze LoRA Fine-tuning and DeepSpeed Configurations

### ISSUE THREAD

**Issue by Gayanukaa** [EdgeVLM-Labs/videollama3-adaptation#4]
Check LoRA fine-tuning as an alternative to full fine-tuning. LoRA significantly reduces memory usage and training time by only training low-rank adapter layers while keeping the base model frozen. This will enable:

#### Phase 1: LoRA Configuration Optimization

- [ ] **Create LoRA configuration script**

  - File: `scripts/finetune/finetune_qved_lora.sh`
  - Add LoRA parameters:
    ```python
    --use_lora True
    --lora_r 16  # Rank
    --lora_alpha 32  # Scaling factor
    --lora_dropout 0.05
    --lora_target_modules "q_proj,v_proj,k_proj,o_proj"  # Attention layers
    --lora_bias "none"
    ```

- [ ] **Modify training script to support LoRA**

  - Update `videollama3/train.py` to detect `--use_lora` flag
  - Apply LoRA to specified modules
  - Ensure only LoRA parameters are trainable

- [ ] **Update model saving/loading**
  - Save only LoRA adapter weights
  - Document merge process for deployment

#### Phase 2: DeepSpeed Configuration Optimization

- [ ] **Check ZeRO-3 config for LoRA**

  - File: `scripts/zero3_lora.json`
  - Optimize for parameter-efficient training
  - Enable CPU offloading if needed

- [ ] **Experiment with DeepSpeed features**

  - Test gradient checkpointing levels
  - Try activation checkpointing
  - Enable gradient accumulation fusion
  - Test mixed precision (FP16 vs BF16)

- [ ] **Create configuration comparison**
  - ZeRO-2 vs ZeRO-3 with LoRA
  - Memory usage analysis
  - Training speed comparison
